### PR TITLE
fix: set link content for images and pdf, links scaling for images, view content for links without addrs

### PR DESCRIPTION
**Pull request by NikitaZotov**

### CODE REVIEW

File: components/scg/src/scg-model-objects.js [hunk 1]
```diff
@@ -392,28 +391,9 @@ SCg.ModelLink.prototype.getConnectionPos = function (from, dotPos) {
  */
 SCg.ModelLink.prototype.setContent = function (content, contentType) {
     this.content = content;
-    this.contentType = contentType ? contentType : 'string';
+    this.contentType = contentType;
```
Comment (MksmOrlov) [inline components/scg/src/scg-model-objects.js:394]: There was a default type (string) for links.
image
image

File: components/scg/src/scg-render.js [hunk 2]
```diff
@@ -397,44 +398,47 @@ SCg.Render.prototype = {
...
+                d.content = linkDiv.find('.impl').html();
+            }
+            if (!linkDiv.find('.impl').html().length) {
+                linkDiv.find('.impl').html(d.content)
             }
 
-            var g = d3.select(this).attr("style", 'opacity: ' + d.opacityElem + '; stroke: ' + d.strokeElem + '')
-
+            const imageDiv = linkDiv.find('img');
+            const pdfDiv = linkDiv.children().find('canvas');
```
Comment (MksmOrlov) [inline components/scg/src/scg-render.js:421]: A very small and cutted pdf window
image

Comment (MksmOrlov) [inline components/scg/src/scg-render.js:421]: Can't double click on pdf link

Comment (MksmOrlov) [inline components/scg/src/scg-render.js:421]: Synchronize button is stucked after creating the second link
image

Comment (MksmOrlov) [inline components/scg/src/scg-render.js:421]: pdf is flashing while updating

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

File: components/scg/src/scg-render.js [hunk 3]
```diff
@@ -397,44 +398,47 @@ SCg.Render.prototype = {
...
+            let g = d3.select(this).attr("style", 'opacity: ' + d.opacityElem + '; stroke: ' + d.strokeElem + '');
```
Comment (MikhailSadovsky) [inline components/scg/src/scg-render.js:422]: pdf not displayed for format_pdf in SCg without flushing
image

Comment (MikhailSadovsky) [reply components/scg/src/scg-render.js:422]: After flushing I can't open any other concept
image

File: components/scg/src/scg-render.js (hunk 2 above)
Comment (MikhailSadovsky) [reply components/scg/src/scg-render.js:421]: Same small window for large images
image
View after flush:
image

File: components/scg/src/scg-struct.js [hunk 4]
```diff
@@ -62,38 +62,40 @@ function ScgFromScImpl(_sandbox, _editor, aMapping) {
 
                 let newMainNode = editor.scene.getObjectByScAddr(addr);
                 if (newMainNode) {
```
Comment (MikhailSadovsky) [inline components/scg/src/scg-struct.js:64]: Looks like issue that's not related to this pr, but should be fixed:
different thickness of sc-arc on edit mode and view mode
image

File: components/scg/src/scg-struct.js [hunk 5]
```diff
@@ -415,74 +416,34 @@ function scgScStructTranslator(_editor, _sandbox) {
...
-                            keynode = window.scKeynodes['format_png'];
+                        let format = window.scKeynodes['format_txt'];
+
+                        if (link.contentType === 'image') {
+                            format = window.scKeynodes['format_png'];
                         } else if (link.contentType === 'html') {
-                            data = link.fileReaderResult;
-                            type = sc.ScLinkContentType.String;
-                            keynode = window.scKeynodes['format_html'];
+                            format = window.scKeynodes['format_html'];
```
Comment (MikhailSadovsky) [inline components/scg/src/scg-struct.js:440]: SCg view for `"format_html"` currently became too slow. It was really fast previously..

**Review by MikhailSadovsky — CHANGES_REQUESTED** [review]

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
check unresolved

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
Check unresolved and fix broken links after synchronization with/without type choosing
image

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
Check unresolved

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
image
Jpeg picture is broken after synchronization

**Review by MikhailSadovsky — APPROVED** [review]

**Review by MksmOrlov — APPROVED** [review]

**Pull request merged**